src/platform/packages/shared/kbn-object-utils/src/flatten_object.test.ts

```diff
@@ -29,4 +29,25 @@ describe('flattenObject', () => {
       beta: 3,
     });
   });
+
+  it('should flat nested object properties as last ones (more nested ovverride already flattened)', () => {
```

Comment:
```suggestion
  it('should give nested object properties precedence over already flattened entries', () => {
```

Maybe something like this could be clearer? Otherwise nit on a typo in `ovveride`